flutter/lib/src/flutter_enricher.dart

```diff
@@ -0,0 +1,185 @@
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:sentry/sentry.dart';
+
+typedef WidgetBindingGetter = WidgetsBinding Function();
+
+/// Enriches [SentryEvent]s with various kinds of information.
+/// FlutterEnricher only needs to add information which aren't exposed by
+/// the Dart runtime.
+class FlutterEnricher implements Enricher {
+  FlutterEnricher(
+    this._checker,
+    this._dartEnricher,
+    this._getWidgetsBinding,
+  );
+
+  factory FlutterEnricher.simple(PlatformChecker checker) {
+    return FlutterEnricher(
+      checker,
+      Enricher(),
+      () => WidgetsFlutterBinding.ensureInitialized(),
```

Comment:
I prefer to not call `WidgetsFlutterBinding.ensureInitialized()` here, this could lead to bugs like https://github.com/getsentry/sentry-dart/issues/263
I'd rather prefer that our event processor is executed once the event is captured and then we're pretty sure that the bindings are already inited.

Reply:
That's exactly the reason it's a callback. It just gets called when an event is captured and not at the time of initializing the `FlutterEnricher`.

Reply:
the idea is that this is not even called, the SDK init. already takes care of `WidgetsFlutterBinding.ensureInitialized`, so at this point, you don't need to take care of it.

Reply:
Yeah, but I need the `WidgetsBinding` which is returned from that call. If `FlutterWidgetsBinding` is already initialized it just returns the same instance.

See https://api.flutter.dev/flutter/widgets/WidgetsFlutterBinding/ensureInitialized.html